### PR TITLE
fix(ec2): Add ARN to key pair

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Follow up to https://github.com/cloudquery/cloudquery/pull/1325. This PR also updates `gen.hcl` to ensure there's no drift between the autogenerated code and the template file.

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Test locally on your own infrastructure
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation)) 
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: plugins/source/aws/resources/services/ec2/gen.hcl [hunk 1]
```diff
@@ -252,41 +252,63 @@ resource "aws" "ec2" "instance_types" {
...
-    description       = "The tags assigned to the key pair."
+    type        = "json"
+    description = "Any tags assigned to the key pair."
+    resolver "resolveTags" {
+      path = "github.com/cloudquery/cloudquery/plugins/source/aws/client.ResolveTags"
+    }
+  }
+
+  column "public_key" {
+    skip = true
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ec2/gen.hcl:312]: Both `key_type` and `public_key` were missing in the original PR. I only added `key_type` as `public_key` might bloat the database

File: plugins/source/aws/resources/services/ec2/key_pairs.go [hunk 2]
```diff
@@ -4,23 +4,29 @@ import (
...
-		DeleteFilter:  client.DeleteAccountRegionFilter,
-		Options:       schema.TableCreationOptions{PrimaryKeys: []string{"key_pair_id"}},
-		IgnoreInTests: true,
+		Name:         "aws_ec2_key_pairs",
+		Description:  "Describes an EC2 Key Pair.",
+		Resolver:     fetchEc2KeyPairs,
+		Multiplex:    client.ServiceAccountRegionMultiplexer("ec2"),
+		IgnoreError:  client.IgnoreAccessDeniedServiceDisabled,
+		DeleteFilter: client.DeleteAccountRegionFilter,
+		Options:      schema.TableCreationOptions{PrimaryKeys: []string{"arn"}},
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ec2/key_pairs.go:22]: Made `arn` primary key. The `arn` is based on the key name and one cannot create duplicate names (I checked)

File: plugins/source/aws/resources/services/ec2/key_pairs.go [hunk 3]
```diff
@@ -77,3 +89,8 @@ func fetchEc2KeyPairs(ctx context.Context, meta schema.ClientMeta, parent *schem
 	res <- output.KeyPairs
 	return nil
 }
+func resolveEc2keyPairArn(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	return client.ResolveARNWithRegion(client.EC2Service, func(resource *schema.Resource) ([]string, error) {
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ec2/key_pairs.go:93]: Not sure if there's a better way to do this at the moment while keeping `gen.hcl` and the code from drifting

File: plugins/source/aws/resources/services/ec2/gen.hcl [hunk 4]
```diff
@@ -252,41 +252,63 @@ resource "aws" "ec2" "instance_types" {
...
+    type              = "string"
+    description       = "The Amazon Resource Name (ARN) of the Key Pair"
+    generate_resolver = true
   }
 
   userDefinedColumn "account_id" {
     description = "The AWS Account ID of the resource."
     type        = "string"
     resolver "resolveAWSAccount" {
-      path = "github.com/cloudquery/cq-provider-aws/client.ResolveAWSAccount"
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ec2/gen.hcl:275]: I missed these renames in the previous PR

**Review by candiduslynx — APPROVED** [review]

**Review by disq — APPROVED** [review]

**Pull request merged**